src/Umbraco.Web/Security/WebSecurity.cs

```diff
@@ -74,29 +74,29 @@ private BackOfficeSignInManager SignInManager
 
         private BackOfficeUserManager<BackOfficeIdentityUser> _userManager;
         protected BackOfficeUserManager<BackOfficeIdentityUser> UserManager
-            => _userManager ?? (_userManager = _httpContext.GetOwinContext().GetBackOfficeUserManager());
+            => _userManager ?? (_userManager = _httpContextAccessor.HttpContext.GetOwinContext().GetBackOfficeUserManager());
 
         [Obsolete("This needs to be removed, ASP.NET Identity should always be used for this operation, this is currently only used in the installer which needs to be updated")]
         public double PerformLogin(int userId)
         {
-            var owinCtx = _httpContext.GetOwinContext();
+            var owinCtx = _httpContextAccessor.HttpContext.GetOwinContext();
             //ensure it's done for owin too
             owinCtx.Authentication.SignOut(Constants.Security.BackOfficeExternalAuthenticationType);
 
             var user = UserManager.FindByIdAsync(userId).Result;
 
             SignInManager.SignInAsync(user, isPersistent: true, rememberBrowser: false).Wait();
 
-            _httpContext.SetPrincipalForRequest(owinCtx.Request.User);
+            _httpContextAccessor.HttpContext.SetPrincipalForRequest(owinCtx.Request.User);
 
             return TimeSpan.FromMinutes(_globalSettings.TimeOutInMinutes).TotalSeconds;
         }
 
         [Obsolete("This needs to be removed, ASP.NET Identity should always be used for this operation, this is currently only used in the installer which needs to be updated")]
         public void ClearCurrentLogin()
         {
-            _httpContext.UmbracoLogout();
-            _httpContext.GetOwinContext().Authentication.SignOut(
+            _httpContextAccessor.HttpContext.UmbracoLogout();
```

Comment:
The one part about using httpcontextaccessor and umbracocontextaccessor is that the result might be null and we are no doing null checks where necessary but we should be. This could be encapsulated in a property getter with a null check. There's probably a few places where these things need to be fixed up.

Reply:
I inserted a null check in the property `AspNetHttpContextAccessor.HttpContext`.